momento-sdk/src/main/java/momento/sdk/SubscriptionWrapper.java

```diff
@@ -70,6 +84,16 @@ public void onError(Throwable t) {
               firstMessage = false;
               future.completeExceptionally(t);
 
+              // Retry logic for UNAVAILABLE errors
+              if (t instanceof io.grpc.StatusRuntimeException) {
+                io.grpc.StatusRuntimeException statusRuntimeException =
+                    (io.grpc.StatusRuntimeException) t;
+                if (statusRuntimeException.getStatus().getCode() == Status.Code.UNAVAILABLE) {
+                  logger.info("Retrying subscription after a delay...");
```

Comment:
i think it's probably reasonable to log this at `info`, but if you are going to do that, you need to make sure log something else earlier that shows that an error occurred or whatever, so that users won't just see this log message and wonder "why?"